frontend/src/store/enrolmentReport.js

```diff
@@ -11,6 +14,12 @@ export const useEnrolmentReport = defineStore('enrolmentReport', {
   actions: {
     async checkDueReports(organizationId, programYearId) {
       const appStore = useAppStore();
+      const authStore = useAuthStore();
+
+      if (!authStore.hasPermission(PERMISSIONS.SUBMIT_ENROLMENT_REPORT)) {
+        return false;
+      }
+
```

Comment:
While I appreciate the proactive check here, I don't think we need this as it starts to blur the lines between business logic and security logic.